subsys/bluetooth/mesh/blob_cli.c

```diff
@@ -969,7 +979,17 @@ static void chunk_send_end(struct bt_mesh_blob_cli *cli)
 
 	cli->chunk_idx = next_missing_chunk(cli, cli->block.missing, cli->chunk_idx + 1);
 	if (cli->chunk_idx < cli->block.chunk_count) {
-		chunk_send(cli);
+		if (cli->chunk_interval_ms > 0) {
+			const struct blob_cli_broadcast_ctx ctx = {
+				.send = chunk_tx,
+				.next = next_chunk_delayed_send,
+				.acked = false,
+			};
+
+			blob_cli_broadcast(cli, &ctx);
```

Comment:
I guess due to forcing to `blob_cli_broadcast` functionality without `chunk_send` device looses ability to be suspended, looses checking ability of emergently ending transaction as well as never gets `BT_MESH_BLOB_CLI_STATE_BLOCK_SEND` state.

Reply:
When it is here, it is already in the `BT_MESH_BLOB_CLI_STATE_BLOCK_SEND` state(because we always first get into from state `chunk_send`), need to look into suspension and checking ability of emergently ending transaction still. Since whether we are at `chunk_send` or `chunk_send_end` we are always in state  `BT_MESH_BLOB_CLI_STATE_BLOCK_SEND`, and the initial call (without my changes) of  `chunk_send_end` is also via workqueue and `blob_cli_broadcast`, I had assumed with this new state on disruption or failure it would be handled the same.

Reply:
Regarding suspension: If I understand it correctly from spec, we are not able / should not be able to suspend during chunk sending anyways (only when we are sending the block). Good depiction of that can be found in MBT model spec 5.3
And I am not sure now what you meant exactly by "checking ability of emergently ending transaction", could you point me to spec/code part you mean by that?

Reply:
What I meant in my question. If we take a look at the `send_chunk` function that is called periodically to send chunks, we will see:
```
static void chunk_send(struct bt_mesh_blob_cli *cli)
{
	struct blob_cli_broadcast_ctx ctx = {
		.send = chunk_tx,
		.next = chunk_send_end,
		.acked = false,
	};

	if (cli->xfer->mode == BT_MESH_BLOB_XFER_MODE_PULL) {
		ctx.send_complete = chunk_tx_complete;
	}

	if (!targets_active(cli)) {
		if (targets_timedout(cli)) {
			suspend(cli);
			return;
		}

		end(cli, false);
		return;
	}

	LOG_DBG("%u / %u size: %u", cli->chunk_idx + 1, cli->block.chunk_count,
		chunk_size(cli->xfer, &cli->block, cli->chunk_idx));

	cli->state = BT_MESH_BLOB_CLI_STATE_BLOCK_SEND;
	blob_cli_broadcast(cli, &ctx);
}
```
It checks which target is active and if the target isn't active anymore it skips sending chunks further.
If target has time out, it goes into suspension.
Calling `blob_cli_broadcast` directly, we skip all these checking. That means if some conditions occur for a device that is sent chunks with delay, then behavior will be different for these conditions.
Per my understanding, new API ant its implementation should add sending with delay but not change behavior for conditions.

Reply:
I see what you mean. Well in that case, I could either insert the same target_active checks within chunk_send_end (in case of chunk interval set) or I could change the approach for this to call chunk_send_end delayed (from chunk_send), removing the next_chunk_delayed_send state again.
diagram wise the change I am suggesting for this would be to have following instead:
chunk_send --|chunk interval as delay|--> chunk_send_end

Or do you have other suggestions?

Reply:
(though I'm not Alex I'll try to reply) I think if you follow the suggestion from https://github.com/zephyrproject-rtos/zephyr/pull/77029#discussion_r1727013411 (unless I miss something) you won't need to worry about this as the delay functionality will be hidden under the broadcast API and the client logic will stay the same.

Reply:
I changed it now, please take a look if this is how you've imagined it :)

Reply:
Why not to add the the delay to `struct blob_cli_broadcast_ctx` like I suggested initially? I meant this (`blob_mdls` tests passing with 0ms, 100ms and 500ms in `chunk_send`): https://github.com/PavelVPV/zephyr/commit/96c6363d79927bed31748cc940c4eb66fd255833

Reply:
Didn't see this option :) Will see to amend.